File: druthvik/tutorial/server/src/models/category.ts
import {
  Column,
  Entity,
  JoinColumn,
  JoinTable,
  ManyToMany,
  ManyToOne,
  OneToMany,
  PrimaryGeneratedColumn,
} from 'typeorm';
import { Tutorial } from './tutorial';

@Entity()
export class Category {
  @PrimaryGeneratedColumn()
  id!: number;

  @Column()
  title?: string;

  @ManyToMany((_type) => Tutorial, (tutorial) => tutorial.categories)
  tutorials!: Tutorial[];
}

File: harsha/Tutorial/server/src/main/java/com/example/postgresdemo/model/Category.java
package com.example.postgresdemo.model;

import java.util.List;
import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToMany;
import javax.persistence.Table;
import javax.validation.constraints.NotBlank;
import org.springframework.web.bind.annotation.CrossOrigin;
import com.fasterxml.jackson.annotation.JsonBackReference;

@Entity
@Table(name = "categories")
@CrossOrigin
public class Category {

	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private int id;

	@NotBlank
	private String title;

	@JsonBackReference
	@ManyToMany( mappedBy = "categories")
	private List<Tutorial> tutorials;

	public Category() {
	}

	public Category(int id, @NotBlank String title, List<Tutorial> tutorials) {
		this.id = id;
		this.title = title;
		this.tutorials = tutorials;
	}

	public List<Tutorial> getTutorials() {
		return tutorials;
	}

	public void setTutorials(List<Tutorial> tutorials) {
		this.tutorials = tutorials;
	}

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public String getTitle() {
		return title;
	}

	public void setTitle(String title) {
		this.title = title;
	}
}
